### DISCUSSION TITLE
Set Axis Title Based on Extras

### DISCUSSION THREAD

**Discussion by baum-sparton** [patrykandpatrick/vico#931] [Ideas]
### Question

Hi,

Thanks again for a great library.

I'm looking to modify the axis title during a transaction.  In particular, my y-axis title includes the units for the data being displayed, and the user has the ability to customize the units used.

My initial thought was to leverage extras, but `CartesianChart.startAxis `does not take a callback with provides either `ExtraStore `or `CartesianChartModel`.

Any suggestions?  Thanks

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by baum-sparton** [comment]
I'm just pinging as I encountered this issue again.  I have a graph where the units are adjusted based on the domain of the data.  Is there anyway to update the axis title based on the extras or am I looking at implementing a custom axis based off of HorizontalAxis or VerticalAxis?  Thanks

**Comment by Gowsky** [comment]
Hello, @baum-sparton. At the moment, `HorizontalAxis` and `VerticalAxis` don't offer a built-in solution for updating the title based on extras. The type of the `title` parameters is a plain `String`, not an `ExtraStore` lambda. However, we plan on adding the capability in question, so I'm moving this discussion to "Ideas."

In the meantime, you can achieve this by subclassing, as per your comment.

Another option is to ignore the synchronization rule and update the title based on the data from the latest transaction. This would involve introducing a state variable and updating its value after calling `runTransaction`. As described in the documentation, this generally isn't recommended, but in this particular case, the only potential consequence is that the title will change a few milliseconds earlier than the host receives the new data. This is unlikely to be noticeable, especially if the chart is animated. However, if this is a concern, subclassing and implementing an `ExtraStore` lambda will provide perfect synchronization.

**Reply by baum-sparton** [reply]
@Gowsky We did end up writing a subclass to allow us to pass the axis titles via the ExtraStore and it does appear to be working good.  I do think this is a situation likely to arise with other users, so moving to Ideas makes sense.  Thanks.

**Comment by Gowsky** [comment]
[Vico v3.0.0-alpha.3](https://github.com/patrykandpatrick/vico/releases/tag/v3.0.0-alpha.3) resolves this.

**Comment by patrickmichalik** [comment]
[Vico 3.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v3.0.0) is now available.
